app/controllers/PointController.scala

```diff
@@ -0,0 +1,95 @@
+package controllers
+
+import common.{Admin, BadRequest, Driver}
+import dao.SessionDao
+import play.api.mvc.{AnyContent, MessagesControllerComponents, Request}
+import security.UserAction
+import services.RouteService
+import common.Serialization.{INSTANCE => Json}
+import javax.inject.{Inject, Singleton}
+import scala.jdk.CollectionConverters._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PointController @Inject()(routeService: RouteService,
+                                sessionDao: SessionDao,
+                                cc: MessagesControllerComponents,
+                                override val userAction: UserAction
+                               )(implicit ec: ExecutionContext)
+  extends SecuredController(cc, userAction) {
+
+  def createPoint() = securedAsync(Admin.INSTANCE :: Nil, Action.async {
+    request: Request[AnyContent] => {
+      logger.info("Received postponed location update request")
+      val body = request.body.asText
+      body match {
+        case Some(bodyString) =>
+          val point = Json.fromJson(bodyString, classOf[common.Point])
+          routeService.createPoint(point).map {
```

Comment:
Why does `routeService` create a `Point`?
I expected that `routeService` would only create `RoutePoint`s...

Reply:
since they have to be in same repo it's ok to propagate this to DAO/Serivces while those are not huge

Reply:
hmmm ok